src/capi.ts

```diff
@@ -48,8 +51,27 @@ const articleContributors = (content: Content): Tag[] =>
 const isImage = (elem: BlockElement): boolean =>
     elem.type === ElementType.IMAGE;
 
+const isVideo = (elem: BlockElement): boolean =>
+    elem.type === ElementType.CONTENTATOM &&
+    elem.contentAtomTypeData?.atomType === "media";
+
 const articleMainImage = (content: Content): Option<BlockElement> =>
-    fromNullable(content?.blocks?.main?.elements.filter(isImage)[0]);
+    fromNullable(content?.blocks?.main?.elements.filter(isImage)[0])
+
+const articleMainVideo = (content: Content): Option<BlockElement> =>
+    fromNullable((content?.blocks?.main?.elements.filter(isVideo) ?? [])[0]);
+
+const articleMainMedia = (content: Content, context: Context): MainMedia =>
```

Comment:
Would it make sense to have this return `Option<MainMedia>`? So the type becomes:

```ts
type MainMedia =
     { kind: MainMediaKind.Image; image: ImageData } |
     { kind: MainMediaKind.Video; video: VideoData }
```

Either we have an image main media with all the image data to render it, or we don't (likewise with video). That way you can simplify some of the logic in `headerMedia.tsx`, because once you unwrap the `Option` you *know* you have the `VideoData` or `ImageData`.